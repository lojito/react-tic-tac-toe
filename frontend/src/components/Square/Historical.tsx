import React, { FC, memo } from "react";
import defaultImage from "../../assets/images/default.jpg";
import { SquareBoard } from "../../types";
import "./Square.scss";

const Square: FC<SquareBoard> = memo(
  ({ disabled, id, imagePath, className }) => {
    let url;
    if (imagePath === "default") {
      url = `url(${defaultImage}`;
    } else {
      url = `url(${process.env.PUBLIC_URL}/images/${imagePath}.jpg)`;
    }

    return (
      <button
        className={className}
        disabled
        id={id}
        style={{ backgroundImage: url }}
        aria-label="historical"
      />
    );
  }
);

export default Square;
