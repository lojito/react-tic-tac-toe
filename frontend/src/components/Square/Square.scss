.board {
    button {
      width: 33%;
      border: 1px solid black;
      display: inline-block;
      background-size: 100% 100%;
      
      &:after {
        content: "";
        display: block;
        padding-bottom: 100%;
      }
  
      &:not([disabled]) {
        cursor: pointer;
      }
    }
  }

  
  .win {
    animation: win 1s 3;
    opacity: 0.7;
  }
  
  @keyframes win {
    from {
      opacity: 1;
    }
    to {
      opacity: 0.7;
    }
  }
  