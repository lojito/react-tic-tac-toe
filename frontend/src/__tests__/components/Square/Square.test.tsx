import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import Square from "../../../components/Square/Square";

describe("Square", () => {
  const renderSquare = (imagePath = "default") => {
    const utils = render(
      <Square disabled id="100" imagePath={imagePath} className="win" />
    );
    const squareButton = screen.getByRole("button", { name: "square" });

    return { ...utils, squareButton };
  };

  it("should render correctly", () => {
    const { asFragment } = renderSquare();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should apply the props", () => {
    const imagePath = "soccer/6";

    const { squareButton } = renderSquare(imagePath);

    expect(squareButton).toBeDisabled();
    expect(squareButton).toHaveAttribute("id", "100");
    expect(squareButton).toHaveAttribute("class", "win");
    expect(squareButton).toHaveStyle({ backgroundImage: /soccer\/6/ });
  });
});
