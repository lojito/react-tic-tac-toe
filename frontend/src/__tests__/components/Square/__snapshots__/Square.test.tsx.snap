// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="square"
    class="win"
    disabled=""
    id="100"
    style="background-image: url(test-file-stub);"
  />
</DocumentFragment>
`;
